mas-foundation/src/androidTest/java/com/ca/mas/foundation/MASPerformanceTest.java

```diff
@@ -0,0 +1,662 @@
+
+        TestId testId = new Object() {}.getClass().getEnclosingMethod().getAnnotation(TestId.class);
+
+        int id = testId.value();
+        ScenarioInfo scenarioInfo = map.get(id);
+        Double sum = 0.0;
+        MASRequest request = new MASRequest.MASRequestBuilder(new URI(GatewayDefaultDispatcher.OTP_PROTECTED_URL)).build();
+
+        MAS.start(getContext());
+        MAS.setGrantFlow(MASConstants.MAS_GRANT_FLOW_PASSWORD);
+        MASCallbackFuture<MASUser> callback = new MASCallbackFuture<>();
+        MASUser.login("admin", "7layer".toCharArray(), callback);
+
+        callback.get();
+        for (int i = 0; i < scenarioInfo.getIteration(); i++) {
+            final CountDownLatch countDownLatch = new CountDownLatch(1);
+            long start = System.currentTimeMillis();
+
+
+            MAS.setAuthenticationListener(new MASAuthenticationListener() {
+
+                @Override
+                public void onAuthenticateRequest(Context context, long requestId, MASAuthenticationProviders providers) {
+
+                }
+
+                @Override
+                public void onOtpAuthenticateRequest(Context context, final MASOtpAuthenticationHandler handler) {
+                    handler.deliver("EMAIL", new MASCallback<Void>() {
+                        @Override
+                        public void onSuccess(Void result) {
+                            handler.proceed(getContext(), "1234");
+                        }
+
+                        @Override
+                        public void onError(Throwable e) {
+
+                        }
+                    });
+                }
+            });
+
+            MAS.invoke(request,new MASCallback<MASResponse<JSONObject>>()
+            {
+                @Override
+                public void onSuccess (MASResponse < JSONObject > result) {
+                    Log.d(TAG, result.getResponseMessage());
+                    countDownLatch.countDown();
+                }
+
+                @Override
+                public void onError (Throwable e){
+                    countDownLatch.countDown();
+                }
+            });
+
+
+            try {
+                countDownLatch.await();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+
+            long end = System.currentTimeMillis();
+
+            sum = sum + (end - start);
+            Log.d(TAG, "Duration of get flow for iteration " + i + " = " + (end - start)/(double) TENS + "s");
+        }
+        double avg = sum / (scenarioInfo.getIteration()* TENS);
+        if(isBenchmark)  {
+            scenarioInfo.setBenchmark(avg);
+        }
+        Log.d(TAG, "Benchmark = " + avg + "s");
+        assertTrue("Taken more than " +scenarioInfo.getBenchmark() +" time to execute", avg <= scenarioInfo.getBenchmark());
+
+    }
+
+    private static void printUpdatedBenchmark() {
+
+        Gson gson = new GsonBuilder().create();
```

Comment:
for writing the result of the benchmark take a look into this method android.content.Context#openFileOutput

Reply:
This API will not work for our requirement of having a file readable and writable programmatically and manually.